auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/EntitlementService.java

```diff
@@ -183,6 +184,10 @@ public boolean datalakeLoadBalancerEnabled(String accountId) {
         return isEntitlementRegistered(accountId, CDP_DATA_LAKE_LOAD_BALANCER);
     }
 
+    public boolean publicEndpointAccessGatewayEnabled(String accountId) {
```

Comment:
observation: the entitlement flag is not used extensively, that is, a lot of changed functionality cannot be switched off as-is

Reply:
I've added the entitlement check anywhere we're setting the endpoint gateway related fields. If the original environment request has those fields set but the entitlement is disabled, then it will unset them before processing the request. I also added a lot of unit tests to verify the behavior when the entitlement is disabled and enabled.